tests/python/gpu/test_gluon_gpu.py

```diff
@@ -233,6 +236,45 @@ def test_symbol_block_fp16():
             break
     assert np.dtype(net_fp16.params[param_name].dtype) == np.dtype(np.float16)
 
+
+@with_seed()
+def test_large_models():
+    ctx = default_context()
+    # Create model
+    net = gluon.nn.HybridSequential()
+
+    largest_num_features = 256
+    with net.name_scope():
+        net.add(nn.Conv2D(largest_num_features, 3))
+
+    net.hybridize()
+    net.initialize(mx.init.Normal(sigma=0.01), ctx=ctx)
+
+    # Compute the height (=width) of the square tensor of the given size in bytes
+    def tensor_size(big_tensor_bytes):
+        bytes_per_float = 4
+        sz = int(math.sqrt(big_tensor_bytes / largest_num_features / bytes_per_float))
+        return (sz // 100) * 100
+
+    # The idea is to create models with large tensors of (say) 20% of the total memory.
+    # This in the past has given cudnnFind() trouble when it needed to allocate similar I/O's
+    # from the area carved out by the MXNET_GPU_MEM_POOL_RESERVE setting (by default 5%).
+    (free_mem_bytes, total_mem_bytes) = mx.context.gpu_memory_info(ctx.device_id)
+    start_size = tensor_size(0.20 * total_mem_bytes)
+    num_trials = 10
+    sys.stderr.write(' testing global memory of size {} ... '.format(total_mem_bytes))
+    sys.stderr.flush()
+    for i in range(num_trials):
+        sz = start_size - 10 * i
```

Comment:
I'm not 100% sure but is there possibly a bug here?  start_size is roughly 20% of the available number of bytes on the GPU right?  It seems like we want to do a few trials and gradually increase I/O tensor sizes but we're subtracting -10 * i.  Is this what we want or do we want to start with 20% of the gpu's allocation and gradually work down?  I.e. something like?

```python
sz = start_size  * ( (10 - i)  / 10)
```

Reply:
We don't want to test the same size, since that would be a hit in the algo cache and cudnnFind() would not be re-run.  The idea is to test different sizes that are roughly the size as chosen initially.  I observed that with the progression of these models, the MXNet storage manager would slowly allocate up to 95% of memory by default (although much of it ends up in its free store).  At that point, if MXNet's storage manager is asked for a big allocation that doesn't match the size of something in the free store, it will trigger a beneficial ReleaseAll():
```
    if (free <= total * reserve_ / 100 || size > free - total * reserve_ / 100)
      ReleaseAll();
```
However, if instead cudnnFind is called at that point, the large allocations it wants to make for I/Os and workspace will cause an out of memory or 'no algo found' error, despite the fact that unused allocations are in the free store.  I'm using the existing Alloc/DirectFree API to possibly trigger a ReleaseAll() prior to calling cudnn within the algo-setting callback.

Reply:
And to get back to your question more succinctly, there's nothing magic about the initial 1200x1200 starting point.  It was calculated based on the 8GB global memory of the CI machines.  I verified that the test initially failed, then passed, on 16GB and 32GB GPUs (with different sizes in each case).

Reply:
Ah that makes total sense, you're just using dimensions to make sure we're not pulling the alg straight from the registry and bypassing the auto-tuning.  Gotcha, thanks Dick.